### DISCUSSION TITLE
invalid frequency error while using DeepAREstimator on data created using PandasDataset

### DISCUSSION THREAD

**Discussion by mohammednaseefop** [awslabs/gluonts#2641] [Q&A]
I'm using PandasDataset to create the model ready data as follows

```
from gluonts.dataset.pandas import PandasDataset
ds = PandasDataset.from_long_dataframe(df, target="Expense", item_id="Property ID")

```
ds.freq returns 'AS-JAN' as my frequency. 

later when I try the following code I'm getting an error
```
from gluonts.mx import DeepAREstimator, Trainer

estimator = DeepAREstimator(
    freq=ds.freq, prediction_length=2, trainer=Trainer(epochs=1)
)
predictor = estimator.train(ds)
predictions = predictor.predict(ds)
```
```
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
[<ipython-input-128-26b28388962d>](https://localhost:8080/#) in <module>
      1 from gluonts.mx import DeepAREstimator, Trainer
      2 
----> 3 estimator = DeepAREstimator(
      4     freq=ds.freq, prediction_length=2, trainer=Trainer(epochs=1)
      5 )

2 frames
[/usr/local/lib/python3.8/dist-packages/gluonts/time_feature/lag.py](https://localhost:8080/#) in get_lags_for_frequency(freq_str, lag_ub, num_lags)
    141         )
    142     else:
--> 143         raise Exception("invalid frequency")
    144 
    145     # flatten lags list and filter

Exception: invalid frequency
```
Why am I getting the error? I am guessing not all the frequency strings are supported by functions in GluonTS.
How do I fix this issue?

**Comment by jaheba** [comment]
Hey, what is ds.freq?

**Reply by mohammednaseefop** [reply]
I'm following the gluonts [document](https://ts.gluon.ai/dev/tutorials/data_manipulation/pandasdataframes.html) for pandas.Data Frame based dataset with my dataset.
I took the code from there. 
```
ds = PandasDataset.from_long_dataframe(df, target="Expense", item_id="Property ID")
ds.freq
```
ds.freq returns the frequency of the timeseries. In the gluonts document the frequency of the data is hourly so it returns "H".

**Reply by jaheba** [reply]
Sorry, maybe I misread your question initially, but you write:

> ds.freq returns 'AS-JAN' as my frequency.

Now you claim it is ``H``.

---

To me it looks like we don't normalise the frequency. `AS-JAN` should just become `A`, which `get_lags_for_frequency` relies on. As a quick fix, you should be able to just pass `freq="A"` to DeepAR.

**Reply by mohammednaseefop** [reply]
Thanks!
Just to clarify . My custom data's frequency is 'AS-JAN'. In the gluonts document  they are using a data at hourly frequency.

**Reply by jaheba** [reply]
Got it! Let me get the issue fixed in #2645 :)
